internal/pkg/otel/run.go

```diff
@@ -80,3 +86,23 @@ func newSettings(version string, configPaths []string) (*otelcol.CollectorSettin
 		DisableGracefulShutdown: true,
 	}, nil
 }
+
+func ensureRegistryExists() error {
+	storageDir := os.Getenv("STORAGE_DIR")
+	if storageDir == "" {
+
+		// by default use "${path.data}/registry/otelcol" to store offsets
+		storageDir = filepath.Join(paths.Data(), "registry", "otelcol")
+
+		// set the STORAGE_DIR env. This will be used by otel collectore to get the registry directory
+		os.Setenv("STORAGE_DIR", storageDir)
+	}
+	if _, err := os.Stat(storageDir); err == nil {
+		// directory exists
+		return nil
+	} else if os.IsNotExist(err) {
+		return os.MkdirAll(storageDir, 0755)
+	} else {
+		return fmt.Errorf("error stating %s: %w", storageDir, err)
+	}
```

Comment:
We can maybe restructure this block a bit
```suggestion

	err := os.Stat(storageDir);
	if errors.Is(err, fs.ErrNotExist) {
		// create directory if it doesn't exist
		return os.MkdirAll(storageDir, 0755)
	} 
	
	if err != nil  {
		// we have a generic error
		return fmt.Errorf("error stating %s: %w", storageDir, err)
	}
	
	return nil
```